src/table.js

```diff
@@ -343,17 +343,39 @@ export const cloneRowAt = rowIndex => tr => {
       const cloneRow = tableNode.child(rowIndex);
       // Re-create the same nodes with same attrs, dropping the node content.
       let cells = [];
+      let rowWidth = 0;
       cloneRow.forEach(cell => {
-        cells.push(
-          tableNodes[cell.type.spec.tableRole].createAndFill(
-            cell.attrs,
-            cell.marks
-          )
-        );
+        // If we're copying a row with rowspan somewhere, we dont want to copy that cell
+        // We'll increment its span below.
+        if (cell.attrs.rowspan === 1) {
+          rowWidth += cell.attrs.colspan;
+          cells.push(
+            tableNodes[cell.type.spec.tableRole].createAndFill(
+              cell.attrs,
+              cell.marks
+            )
+          );
+        }
       });
 
+      // If a higher row spans past our clone row, bump the higher row to cover this new row too.
+      let newTr = tr;
+      if (rowWidth < map.width) {
+        for (let i = rowIndex; i >= 0; i--) {
+          newTr = forEachCellInRow(i, (cell, tr) => {
+            let rowspan = cell.node.attrs.rowspan;
+            if (rowspan === 1) {
+              return tr;
+            }
+
+            i = -1;
```

Comment:
do we need this?

Reply:
To exit the loop once we've hit the closest parent row with a rowspan.